carmajava/geometry/src/main/java/gov/dot/fhwa/saxton/carma/geometry/cartesian/spatialstructure/NSpatialHashMapFactory.java

```diff
@@ -14,28 +14,29 @@
  * the License.
  */
 
-package gov.dot.fhwa.saxton.carma.geometry.cartesian.spatialhashmap;
+package gov.dot.fhwa.saxton.carma.geometry.cartesian.spatialstructure;
 
 import java.util.HashMap;
 
 import gov.dot.fhwa.saxton.carma.geometry.cartesian.AxisAlignedBoundingBox;
 
 /**
- * Factory/builder pattern class for the construction of NSpatialHashMap instances.
+ * Factory pattern class for the construction of NSpatialHashMap instances.
  * <p>
  * Assigns the strategies needed by the NSpatialHashMap instance.
  */
-public class NSpatialHashMapFactory {
+public class NSpatialHashMapFactory implements ISpatialStructureFactory{
+    public final double[] cellDims;
```

Comment:
I think the goal for factory pattern is to create different classes with the same interface with one factory. So it should make more sense for me to have buildSpatialStructure taking in the dimension and returning a NSpatialHashMap object.

Reply:
The issue is that not every spatial structure would have the same constructor. If we had an octree there wouldn't be a cell size array to pass in. A kdtree would similarly be a different set of parameters.

Reply:
You are right this isn't pure factory pattern (maybe it qualifies as builder). I want dependency injection of the specific type of spatial structure but I need to be able to create multiple structures, thus the interface for the factory instead of just the object

Reply:
Got you.